### PR TITLE
SPP-9926 Add New Packages

### PR DESCRIPTION
**Pull request by sanjeevz3009**
# Description

Currently, across the 2 different repos (sml-catalogue & sml-python-small) we worked with/ are working on, the dev packages installed aren't consistent with each other + aren't in the correct dependency groups.

Code comments & documentation updates are not necessary in this instance.
## Type of change

- Pylint, flake8, bandit, isort and black are in the repo now.
- Running pylint, flake8, isort and black on features/ steps python files as well now
- The linters and code formatter have been run and issues from them have been resolved.
- Code has been changed to adhere to best practices/ resolved issues raised by the linters. 
- Code formatted using black.
- Isort ran on the code.
- Suppressed several linting-related errors within the features/ steps Python files such as code duplication etc as it's the way we write the selenium/ behave tests.
- Sorted the dependencies within pyproject.toml into the correct groups.
- CI pipeline updated to reflect the changes.

**Excluded the features/ steps python files from security scanning due to B101 bandit security vulnerability being raised. We can ignore this low-risk vulnerability in this instance as we are using assert for testing purposes.** 
https://bandit.readthedocs.io/en/latest/plugins/b101_assert_used.html

Another thing to note is when the concourse pipelines run and the jobs pass, it doesn't mean the tests are passing. As long as all the jobs successfully finish the GitHub checks below pass, we may assume the tests are passing but I recently found out that this is not the case. The only way to know is to go to the concourse pipelines and manually scroll to the tests to see if it has failed or passed. So there may have been cases where failing test code may have been merged to main. Considering we test it locally and only push it up when it passes, this may be least likely to have occurred.

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Technical Enhancements

# Checklist:

If any of these are not completed, please explain why in the notes.

## **Definition of Done**
**Code and merges**

- [x] Code to be commented on where applicable 
- [x] Documentation updated where required 
- [x] Have considered non-functional requirements such as Security, Performance, Scalability, and Fault Tolerance
- [x] I have linted the code

**Testing**

- [x] All levels of acceptance test are passing (automated, integration, manual, accessibility, etc.)
- [x] I have run the behave command to check the selenium behaviour tests pass locally
- [x] Acceptance criteria met

**Other Checks**
- [x] I have performed a self-review of my own code
- [x] I have checked for spelling errors
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes don't break anything unexpected
- [x] I have checked and updated the security.txt file where required
- [x] Up to date with the main branch

### CODE REVIEW

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Review by jhmjam — DISMISSED** [review]

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

File: .flake8 [hunk 1]
```diff
@@ -2,4 +2,6 @@
 max-line-length = 160
 extend-ignore =
     # E402 module level import not at top of file
-    E402
\ No newline at end of file
+    E402
+per-file-ignores =
```
Comment (gibbardsteve) [inline .flake8:6]: Add a comment to explain why you are ignoring this

File: features/accessibility.feature [hunk 2]
```diff
@@ -2,12 +2,6 @@
 
 Feature: About library tests
 
-    Scenario: Home page accessibility check
```
Comment (gibbardsteve) [inline features/accessibility.feature:5]: Is there a reason the test has been removed?

File: features/methods.feature [hunk 3]
```diff
@@ -60,5 +60,5 @@ Feature: Methods catalogue tests
     Scenario: Selective editing table row check
         Given I'm an sml portal user trying to get to the methods catalogue page
         When I navigate to the methods catalogue page
-        Then The "ready" table row of the method are "Selective Editing" "Editing" "Editing & Imputation" "Python/Pandas"
+        Then The "ready" table row of the method are "Selectiveee Editing" "Editing" "Editing & Imputation" "Python/Pandas"
```
Comment (gibbardsteve) [inline features/methods.feature:63]: Why has this got 'Selectiveee' ?

File: .flake8 (hunk 1 above)
Comment (sanjeevz3009) [reply .flake8:6]: Flake8 config updated and comments added to it and elsewhere we are ignoring linting errors in-line code, comments added there as well.

File: features/methods.feature (hunk 3 above)
Comment (sanjeevz3009) [reply features/methods.feature:63]: This is due to me testing the concourse pipeline false pass check issue quickly, Jason and I were trying to find a quick fix so decided to test it here. Fixed this and will create a new branch for that ticket which I was working on with Jason.

File: features/accessibility.feature (hunk 2 above)
Comment (sanjeevz3009) [reply features/accessibility.feature:5]: Yes, the linter raised an issue for this test and after speaking with James we decided to remove it. The error was regarding a function/ selenium test being useless so not doing anything. James mentioned how it was to test something or it wasn't really doing anything =)

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Review by gibbardsteve — DISMISSED** [review]

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Comment by ons-spp-machine-user** [comment]
DEPLOY_URL=https://d2jrqb4gejjd5w.cloudfront.net/
cloudfront_id=E3AQAW3XZFG7RX

**Review by gibbardsteve — APPROVED** [review]

**Pull request merged**